### ISSUE TITLE
pyswip ERROR

### ISSUE THREAD

**Issue by anlanxuan** [yuce/pyswip#56]
I tried to install pyswip via Pycharm. It can be installed successfully.

OS: Windows 10 64-bit
IDE: Pycharm 2018.3.2
Language: Python 3.6
Pyswip Version: 0.2.8
I run the test
from pyswip import Prolog
prolog = Prolog()
prolog.assertz("father(michael,john)")

This is my error:
  File "C:/Users/anlanxuan/Desktop/prolog test/venv/Include/prologtest.py", line 1, in <module>
    from pyswip import Prolog
  File "C:\Users\anlanxuan\Desktop\prolog test\venv\lib\site-packages\pyswip\__init__.py", line 29, in <module>
    from pyswip.prolog import Prolog
  File "C:\Users\anlanxuan\Desktop\prolog test\venv\lib\site-packages\pyswip\prolog.py", line 28, in <module>
    from pyswip.core import *
  File "C:\Users\anlanxuan\Desktop\prolog test\venv\lib\site-packages\pyswip\core.py", line 568, in <module>
    (_path, SWI_HOME_DIR) = _findSwipl()
  File "C:\Users\anlanxuan\Desktop\prolog test\venv\lib\site-packages\pyswip\core.py", line 411, in _findSwipl
    (path, swiHome) = _findSwiplWin()
  File "C:\Users\anlanxuan\Desktop\prolog test\venv\lib\site-packages\pyswip\core.py", line 208, in _findSwiplWin
    match = pattern.match(ret[-1])
IndexError: list index out of range

**Comment by janeshin059** [comment]
see this one, it worked for me!
https://github.com/yuce/pyswip/issues/54

**Comment by anlanxuan** [comment]
I have solved this problem，try to use the lower version.

**Comment by wisaaco** [comment]
I've solved modifying the next function (I use MAC):
```
def _findSwiplMacOSHome():
    """
    This function is guesing where SWI-Prolog is
    installed in MacOS via .app.
    
    :parameters:
      -  `swi_ver` (str) - Version of SWI-Prolog in '[0-9].[0-9].[0-9]' format
      
    :returns:
        A tuple of (path to the swipl so, path to the resource file)

    :returns type:
        ({str, None}, {str, None})
    """

    names = ['libswipl.dylib', 'libpl.dylib']
    path  = "/usr/local/Cellar/swi-prolog/8.0.3_1/libexec/lib/swipl/lib/x86_64-darwin/"
    os.environ['SWI_LIB_DIR'] = path
    return (path+names[0],None)
```

**Comment by marcobaturan** [comment]
Dear members,
I found a solution. I had the same mistake but in 64-bit Lubuntu. By deduction and partly by trial and error I found that one has to reinstall the Swi-Prolog package to move to the next version.
Update pip for python3.
And re-install the pyswip package;
link to swi-prolog:

- https://www.swi-prolog.org/build/PPA.html

update pip:

- /usr/bin> sudo -H python3 -m pip install --upgrade pip

Install the latest version of pyswip;

- sudo pip install PySwip==0.2.10

Kind regards,
M.B
